packages/ra-core/src/controller/input/ReferenceArrayInputController.tsx

```diff
@@ -232,7 +234,10 @@ export class UnconnectedReferenceArrayInputController extends Component<
                     'The value of ReferenceArrayInput should be an array'
                 );
             }
-            crudGetMany(reference, ids);
+            const idsToFetch = isInitialCall
```

Comment:
Why dod you need the isInitialCall? The difference should work in all cases.

Reply:
Actually not, when the component just mounted, if we just call `fetchReferences(this.props)`, the difference between the `props` argument of the `fetchReferences` function and `this.props` results in an empty list (they're the same object of course :D). However we still need to call `crudGetMany` with all the values I think

Reply:
I edited this, I think it should be cleaner now